Implementation smell,Namespace,Class,File,Method,Description
Long Identifier,System.Threading.Tasks,TaskHelpersExtensions,C:\repos\owin_museum-piece-gate\src\Samples\Sample.Nancy\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,,The length of the parameter _rethrowWithNoStackLossDelegate is 31.
Long Statement,System.Threading.Tasks,TaskHelpersExtensions,C:\repos\owin_museum-piece-gate\src\Samples\Sample.Nancy\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,CatchImpl,The length of the statement  "                            throw new InvalidOperationException("You must set the Task property of the CatchInfo returned from the TaskHelpersExtensions.Catch continuation."); " is 147.
Long Statement,System.Threading.Tasks,TaskHelpersExtensions,C:\repos\owin_museum-piece-gate\src\Samples\Sample.Nancy\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,CatchImplContinuation,The length of the statement  "            task.ContinueWith(innerTask => tcs.TrySetFromTask(innerTask)' TaskContinuationOptions.NotOnFaulted | TaskContinuationOptions.ExecuteSynchronously); " is 147.
Long Statement,System.Threading.Tasks,TaskHelpersExtensions,C:\repos\owin_museum-piece-gate\src\Samples\Sample.Nancy\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,CatchImplContinuation,The length of the statement  "                                throw new InvalidOperationException("You cannot return null from the TaskHelpersExtensions.Catch continuation. You must return a valid task or throw an exception."); " is 165.
Long Statement,System.Threading.Tasks,TaskHelpersExtensions,C:\repos\owin_museum-piece-gate\src\Samples\Sample.Nancy\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,CatchImplContinuation,The length of the statement  "                            throw new InvalidOperationException("You cannot return null from the TaskHelpersExtensions.Catch continuation. You must return a valid task or throw an exception."); " is 165.
Long Statement,System.Threading.Tasks,TaskHelpersExtensions,C:\repos\owin_museum-piece-gate\src\Samples\Sample.Nancy\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,GetRethrowWithNoStackLossDelegate,The length of the statement  "                // No-ops if the task completed successfully' else throws the originating exception complete with the correct call stack. " is 121.
Empty Catch Block,System.Threading.Tasks,TaskHelpersExtensions,C:\repos\owin_museum-piece-gate\src\Samples\Sample.Nancy\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,GetRethrowWithNoStackLossDelegate,The method has an empty catch block.
Magic Number,Sample.Nancy,DefaultPage,C:\repos\owin_museum-piece-gate\src\Samples\Sample.Nancy\DefaultPage.cs,Invoke,The following statement contains a magic number: if (request.Path == "/")              {                  response.Status = "200 OK";                  response.ContentType = "text/html";                  response.Write("<h1>Sample.App</h1>");                  response.Write("<p><a href='{0}/wilson/'>Wilson</a></p>"' request.PathBase);                  response.Write("<p><a href='{0}/wilsonasync/'>Wilson (async)</a></p>"' request.PathBase);                  response.Write("<p><a href='{0}/nancy/'>Nancy</a></p>"' request.PathBase);                  response.Write("<p><a href='{0}/fileupload'>File Upload</a></p>"' request.PathBase);              }              else              {                  response.StatusCode = 404;              }
Missing Default,System.Threading.Tasks,TaskHelpers,C:\repos\owin_museum-piece-gate\src\Samples\Sample.Nancy\App_Packages\TaskHelpers.Sources.0.2\TaskHelpers.cs,SetIfTaskFailed,The following switch statement is missing a default case: switch (source.Status)              {                  case TaskStatus.Canceled:                  case TaskStatus.Faulted:                      return tcs.TrySetFromTask(source);              }
Missing Default,System.Threading.Tasks,TaskHelpersExtensions,C:\repos\owin_museum-piece-gate\src\Samples\Sample.Nancy\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,CopyResultToCompletionSourceImpl,The following switch statement is missing a default case: switch (task.Status)                  {                      case TaskStatus.Canceled:                      case TaskStatus.Faulted:                          TaskHelpers.TrySetFromTask(tcs' task);                          break;                        case TaskStatus.RanToCompletion:                          tcs.TrySetResult(resultThunk(task));                          break;                  }
Missing Default,System.Threading.Tasks,TaskHelpersExtensions,C:\repos\owin_museum-piece-gate\src\Samples\Sample.Nancy\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,CopyResultToCompletionSourceImplContinuation,The following switch statement is missing a default case: switch (innerTask.Status)                  {                      case TaskStatus.Canceled:                      case TaskStatus.Faulted:                          TaskHelpers.TrySetFromTask(tcs' innerTask);                          break;                        case TaskStatus.RanToCompletion:                          tcs.TrySetResult(resultThunk(task));                          break;                  }
